### DISCUSSION TITLE
Deploy with provisionedConcurrency fail

### DISCUSSION THREAD

**Discussion by jbuitrago-devops-blossom** [serverless/serverless#11887] [Q&A]
### Are you certain it's a bug?

- [X] Yes, it looks like a bug

### Is the issue caused by a plugin?

- [X] It is not a plugin issue

### Are you using the latest v3 release?

- [X] Yes, I'm using the latest v3 release

### Is there an existing issue for this?

- [X] I have searched existing issues, it hasn't been reported yet

### Issue description

When I try to deploy my lambda function with the `provisionedConcurrency: 1`
<img width="502" alt="image" src="https://user-images.githubusercontent.com/79993534/229202665-bce6cbdf-a8b8-44fa-9ae7-f0208eeaafbf.png">
I am getting the following error
<img width="520" alt="image" src="https://user-images.githubusercontent.com/79993534/229202986-3c8482ab-c2f1-4839-aa5c-1cd45c7cc3a6.png">
<img width="697" alt="image" src="https://user-images.githubusercontent.com/79993534/229202912-58ef27d7-6c21-423e-a58a-f625559bd1b9.png">

### Service configuration (serverless.yml) content

```yaml
N/A
```

### Command name and used flags

sls deploy --force --verbose

### Command output

```shell
Error:
CREATE_FAILED: RoutePublicProvConcLambdaAlias (AWS::Lambda::Alias)
Provisioned Concurrency configuration failed to be applied. Reason: FUNCTION_ERROR_INIT_FAILURE
```

### Environment information

```shell
Framework Core: 3.23.0
Plugin: 6.2.3
SDK: 4.3.2
```

**Comment by medikoo** [comment]
@jonatan-blossom error information suggests you have an error in your function, which makes it crash at initialization

Moving this to discussions

**Reply by jbuitrago-devops-blossom** [reply]
When I change `provisionedConcurrency : 0` the deploy works perfectly!

**Reply by cgradwohl** [reply]
Were you able to find more information on this issue? I am facing the same problem.

**Reply by medikoo** [reply]
> When I change provisionedConcurrency : 0 the deploy works perfectly!

That's totally understood, as function is initialized at creation only when provisioned concurrency is turned on

**Reply by imphatic** [reply]
Also having issues.  Add provisionedConcurrency results in error.  Remove it and it works.
